Rework navigation structure

ekohl opened the pull request:
#### What changes are you introducing?

This creates landing pages for each guide where all build flavors are shown. By default they are collapsed because they're fairly long. Then a new individual landing page is created for every build flavor.

Then the navigation is changed so there is a link to the landing page. The version switcher remains in place to allow users to switch between different versions of the guide.

#### Why are you introducing these changes? (Explanation, links to references, issues, etc.)

https://github.com/theforeman/foreman-documentation/pull/3670 started this, but I really liked the quick navigation between guides.

In this PR there is a compromise: it adds a link to a version specific page (`/release/$version/$build`) that contains only the guides for that build without any collapsing. This makes it easy to navigate to other guides in the same build.

#### Anything else to add? (Considerations, potential downsides, alternative solutions you have explored, etc.)

This is incomplete in that it only updated `nightly.json`. Before I spent a lot of time doing that, I'd first like some agreement this is good.

I noticed that it is tedious to switch between builds. Perhaps we should make that easier.

#### Checklists

* [x] I am okay with my commits getting squashed when you merge this PR.
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [ ] Foreman 3.15/Katello 4.17
* [ ] Foreman 3.14/Katello 4.16 (Satellite 6.17)
* [ ] Foreman 3.13/Katello 4.15 (EL9 only)
* [ ] Foreman 3.12/Katello 4.14 (Satellite 6.16; orcharhino 7.2 on EL9 only)
* [ ] Foreman 3.11/Katello 4.13 (orcharhino 6.11 on EL8 only; orcharhino 7.0 on EL8+EL9; orcharhino 7.1 with Leapp)
* [ ] Foreman 3.10/Katello 4.12
* [ ] Foreman 3.9/Katello 4.11 (Satellite 6.15; orcharhino 6.8/6.9/6.10)
* We do not accept PRs for Foreman older than 3.9.

maximiliankolb commented:
Thanks Ewoud, I like it!

I cannot comment much on the Ruby/JS code, but the result looks good.

I have some minor suggestions that could be follow-up PRs:
* show the docs for Katello at the top because they are the most well maintained ones
* Move ACD to "managing hosts"
* Rename "Install Smart Proxy" to "Install Smart Proxy Server" etc.
* To me, "managing content" belongs to administering hosts, not administering F+K
* https://theforeman-foreman-documentation-preview-pr-3915.surge.sh/release/3.14/ is empty. Is that due to surge.sh or because it was not implemented yet? (this would be a blocker to the PR IMHO)
* On the bottom of the index page, I miss a note "Foreman+Katello is a community project. You can contribute to the docs" etc.

Overall :+1:

ekohl commented:
> show the docs for Katello at the top because they are the most well maintained ones

I can live with that.

> Move ACD to "managing hosts"

Done

> Rename "Install Smart Proxy" to "Install Smart Proxy Server" etc.

Done since this matches the document title.

> https://theforeman-foreman-documentation-preview-pr-3915.surge.sh/release/3.14/ is empty. Is that due to surge.sh or because it was not implemented yet? (this would be a blocker to the PR IMHO)

This is because of the TODO I mentioned:

> This is incomplete in that it only updated nightly.json. Before I spent a lot of time doing that, I'd first like some agreement this is good.

Once we add sections it should no longer be empty. I have updated 2 more versions.

> On the bottom of the index page, I miss a note "Foreman+Katello is a community project. You can contribute to the docs" etc.

I'd love it if others could work on that in a follow up. Will others feel comfortable maintaining this?

Now one thing that bothers me is that by default we lead users to nightly guides. I have updated it to link to the latest stable version.
